pkg/controllers/vdb/restart_reconciler.go

```diff
@@ -126,7 +128,7 @@ func (r *RestartReconciler) Reconcile(ctx context.Context, _ *ctrl.Request) (ctr
 	// admin commands to run.  Cluster operations only apply if the entire
 	// vertica cluster is managed by k8s.  We skip that if initPolicy is
 	// ScheduleOnly.
-	if r.PFacts.getUpNodeAndNotReadOnlyCount() == 0 &&
+	if r.PFacts.getUpNodeAndNotReadOnlyCount(r.SandboxName) == 0 &&
```

Comment:
I think we can avoid the filter for most of the podfacts APIs if we collected podfacts for only a single sandbox.